lib/std/child_process.zig

```diff
@@ -393,15 +393,19 @@ pub const ChildProcess = struct {
                 switch (builtin.os.tag) {
                     .linux, .macos, .ios => {
                         var ru: std.os.rusage = undefined;
-                        const res = os.wait4(self.id, 0, &ru);
+                        // see next comment, catch unreachable duplicates
+                        // previous wait4 behavior
+                        const res = os.wait4(self.id, 0, &ru) catch unreachable;
```

Comment:
instead of `catch unreachable` please exhaustively switch on every possible error code and explain why it is unreachable in a comment.